pkg/cmds/root.go

```diff
@@ -39,7 +39,7 @@ func NewRootCmd() *cobra.Command {
 	// a.k.a. change all "_" to "-". e.g. glog package
 	flags.SetNormalizeFunc(cliflag.WordSepNormalizeFunc)
 
-	kubeConfigFlags := genericclioptions.NewConfigFlags(true)
+	kubeConfigFlags := genericclioptions.NewConfigFlags(false)
```

Comment:
Why change this?

Reply:
If set to true, will use persistent client config and propagate the config to the places that need it, rather than loading the config multiple times.

We had to change it because it was taking the namespace as "default" instead of the namespace we provided through -n flag

Reply:
Originally added here: https://github.com/kubernetes/kubernetes/pull/71117

- https://github.com/kubernetes/sample-cli-plugin/blob/master/pkg/cmd/ns.go#L106